numpyro/contrib/autoguide/__init__.py

```diff
@@ -185,8 +174,8 @@ def __call__(self, *args, **kwargs):
         result = {}
 
         for name, unconstrained_value in self._unpack_latent(latent).items():
-            transform = self._inv_transforms[name]
             site = self.prototype_trace[name]
+            transform = biject_to(site['fn'].support)
```

Comment:
We will use `postprocess_fn` to constrain latent samples, so we don't need to store `inv_transforms`. If we don't want to replay model when there are deterministic sites, then we can simply remove that behavior in then `dynamic_args=False` branch of `get_potential_fn` utility. But I think we should record `deterministic` sites.